### ISSUE TITLE
Link error "undefined hidden symbol:" in firefox 93 building with clang

### ISSUE THREAD

**Issue by smsxgli** [openSUSE/firefox-maintenance#37]
I am a KDE user, build firefox 93 with these patches. While during linking `libxul.so`, the lld failed with error
```
ld.lld: error: undefined hidden symbol: nsKDEShellService::IsDefaultForScheme(nsTSubstring<char> const&, bool*)
>>> referenced by nsKDEShellService.cpp
>>>               /build/firefox-kde-opensuse/src/firefox-93.0/obj/toolkit/library/build/../../../browser/components/shell/nsKDEShellService.o:(vtable for nsKDEShellService)
```
I think this might because of the unimplemented virtual function `IsDefaultForScheme`, which should be defined in `browser/components/shell/nsKDEShellService.cpp`, created by the `firefox-kde.patch` file.

I try to fix this error by offering the missing function with just return `NS_ERROR_NOT_IMPLEMENTED`, and then this error is gone. But I have no idea about how to really realize this function.

Build environment:
Arch linux, amd64
Build with clang, llvm and lld, version12

If any more information should be offered, please let me known, and forgive my poor english since I'm not a native speaker.

**Comment by smsxgli** [comment]
@francoism90 I will have a test soon. By the way, I met a link error about LLVM again with your last PKGBUILD and the modified patch.
And I **may not** open a PR for this, because since **only** clang users will meet this issue, and KDE seems **only** support gcc build.
If you want to, please feel free to open a PR for this error.

**Comment by smsxgli** [comment]
This error should be solved in commit `f41e1b2623e162339286610070c059afe5106c25`. Thanks!
